.github/workflows/validate_source_protos.yml

```diff
@@ -0,0 +1,25 @@
+name: Validate Source Proto Files
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  validate:
+    name: Validate Source Proto Files
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v2
+
+    - name: Setup python3
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.7.x'
+
+    - name: Install build dependencies
+      run: python -m pip install -r python_build_requirements.txt
+
+    - name: Validate source proto files against generated
+      run: |
+        python source/codegen/validate_source_protos.py -s source/codegen/protos -g generated
```

Comment:
Should these be `imports/hapigen/protos` instead of `source/codegen/protos`. Maybe `hapigen`-sourced metadata should also be in `imports`. And, either way, could there be a `README.md` in the directory explaining where they came from?

Similarly I'm not getting the name `*_source_protos`. `imported` or `hapigen` would work there. Maybe `dual_sourced`, if that's not too cute.

Reply:
Talked about offline some. Eventually we'll want the imported metadata and proto files (hapigen and scrapigen currently) somewhere outside of the source directory. But for now, we'll keep with the current paradigm and put the hapigen-sourced stuff here in source/codegen/metadata.

I did update all the file names and terminology in the files to use "imported" over "source" to avoid confusion some. Also, added small README.md in each of the hapigen metadata folders to give some insight about where they come from. Also, have a Imported_From_Hapigen.md living in source/codegen/metadata that each of them reference with some more info on what's going on and what to do if the build fails on the "validate imported protos" step.